src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.NoSuchElementException;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int fruitsInStorage = Storage.storage.get(fruit);
+        int fruitsToBuy = fruitTransaction.getQuantity();
+        if (fruitsToBuy > fruitsInStorage) {
+            throw new NoSuchElementException("Not enough fruit for purchase. You can purchase max: "
+                    + fruitsInStorage);
```

Comment:
The `NoSuchElementException` is not the most appropriate here since it's typically used by iterators to indicate that there are no more elements. It's better to use a `RuntimeException` with a clear message to indicate that there is not enough fruit in the storage.